dlib/statistics/statistics.h

```diff
@@ -22,43 +22,24 @@ namespace dlib
     {
     public:
 
-        running_stats()
-        {
-            clear();
-
-            COMPILE_TIME_ASSERT ((
-                    is_same_type<float,T>::value ||
-                    is_same_type<double,T>::value ||
-                    is_same_type<long double,T>::value 
-            ));
-        }
+        static_assert(std::is_floating_point<T>::value, "T must be a floating point type");
 
         void clear()
         {
-            sum_     = 0;
-            sum_sqr  = 0;
-            sum_cub  = 0;
-            sum_four = 0;
-
-            n = 0;
-            min_value = std::numeric_limits<T>::infinity();
-            max_value = -std::numeric_limits<T>::infinity();
+            running_stats empty{};
+            std::memcpy(this, &empty, sizeof(empty));
```

Comment:
This is weird.  Why not `*this = running_stats{};`?

Reply:
Yeah that's better